scan/lib/scan/detect_values.rb

```diff
@@ -175,10 +175,26 @@ def self.detect_destination
 
       # building up the destination now
       if Scan.devices && Scan.devices.count > 0
```

Comment:
Sorry to comment after this has already been merged, but don't the `Scan.project` checks reintroduce [the problem #6637 addresses with single scheme, multi-platform projects](https://github.com/fastlane/fastlane/pull/6637#issue-184110468)? [(Relevant diff here.)](https://github.com/fastlane/fastlane/pull/6637/commits/1b8d6346d68a7e4f88f836f96f426609a2a31de5#diff-08afcb1021a2963d22787f399a0d3b7fL177)

Doesn't look like there were any test cases added to prevent a regression. 

@0xced @mpirri @matthewellis